### PR TITLE
Weekly impact email

### PR DESCRIPTION
**Pull request by vishalmahuli**

### CODE REVIEW

File: core/egov-notification-mail/src/main/resources/application.properties [hunk 1]
```diff
@@ -7,8 +7,8 @@ spring.kafka.consumer.group-id=email
 
 #Email Configuration
 mail.enabled=true
-mail.sender.username=egov-services@egovernments.org
-mail.sender.password=kFjZC5u8lD
+mail.sender.username=egov-micro.services@egovernments.org
```
Comment (nithindv) [inline core/egov-notification-mail/src/main/resources/application.properties]: Reset and remove credentials

File: cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/utils/CronUtils.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	 * 
+	 * @param epochTime
+	 * @return
+	 */
+	public String getDayAndMonth(Long epochTime) {
+		Calendar calendar = Calendar.getInstance();
+		calendar.setTimeInMillis(epochTime);
+		StringBuilder date = new StringBuilder();
+		String suffix = "th ";
+		if((calendar.get(Calendar.DAY_OF_MONTH) % 10) == 1) {
```
Comment (nithindv) [inline cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/utils/CronUtils.java]: Wouldn't this fail, say, for the 11th, 12th and 13th?

File: cron-jobs/egov-weekly-impact-notifier/src/main/resources/templates/welcome.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE HTML>
```
Comment (nithindv) [inline cron-jobs/egov-weekly-impact-notifier/src/main/resources/templates/welcome.html]: Remove the file ?

File: cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronQueries.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
+package org.egov.win.repository;
+
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (nithindv) [inline cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronQueries.java]: Remove component scanning from this file?

File: cron-jobs/egov-weekly-impact-notifier/src/main/resources/velocity/weeklyimpactflasher.vm [hunk 5]
```diff
@@ -0,0 +1,182 @@
+<html>
+	<br>Hi All,</br>
+</br>
+<br>DIGIT is starting to be adopted across the country rapidly through NUS. While this adoption takes place its important for all of us to be aware the extent of its adoption .
```
Comment (nithindv) [inline cron-jobs/egov-weekly-impact-notifier/src/main/resources/velocity/weeklyimpactflasher.vm]: Can we simplify this? For instance, table rows and columns should ideally be generated by looping through a collection.

File: cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronRepository.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+		ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+		Object response = null;
+		try {
+			response = restTemplate.postForObject(uri.toString(), request, Map.class);
+		}catch(HttpClientErrorException e) {
+			log.error("External Service threw an Exception: ",e);
+			throw new ServiceCallException(e.getResponseBodyAsString());
+		}catch(Exception e) {
+			log.error("Exception while fetching from searcher: ",e);
```
Comment (nithindv) [inline cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronRepository.java]: Do we want to return null if we don't get a response?

File: cron-jobs/egov-weekly-impact-notifier/src/main/resources/velocity/weeklyimpactflasher.vm (hunk 5 above)
Comment (vishalmahuli) [reply cron-jobs/egov-weekly-impact-notifier/src/main/resources/velocity/weeklyimpactflasher.vm]: Yes, but need more time to implement, we can send the change in an intermediate release later.

File: cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronRepository.java (hunk 6 above)
Comment (vishalmahuli) [reply cron-jobs/egov-weekly-impact-notifier/src/main/java/org/egov/win/repository/CronRepository.java]: Yes, this is a common repository code we use in all modules, it is written for a generic use-case.

**Pull request merged**